codechecker_lib/instance_manager.py

```diff
@@ -0,0 +1,103 @@
+# -------------------------------------------------------------------------
+#                     The CodeChecker Infrastructure
+#   This file is distributed under the University of Illinois Open Source
+#   License. See LICENSE.TXT for details.
+# -------------------------------------------------------------------------
+"""
+Instance manager handles the state keeping of running CodeChecker instances
+for a particular user on the local machine.
+"""
+
+import getpass
+import json
+import os
+import portalocker
+import psutil
+import stat
+import tempfile
+
+
+def __getInstanceDescriptorPath():
+    return os.path.join(tempfile.gettempdir(), ".codechecker_" +
+                        getpass.getuser() + ".instances.json")
+
+
+def __makeInstanceDescriptorFile():
+    if not os.path.exists(__getInstanceDescriptorPath()):
+        with open(__getInstanceDescriptorPath(), 'w') as f:
+            json.dump([], f)
+        os.chmod(__getInstanceDescriptorPath(), stat.S_IRUSR | stat.S_IWUSR)
```

Comment:
Maybe the code is easier to understand if you only call the `getInstanceDescriptorPath` once.